### PR TITLE
inaugurator: allow cmdline parameters with "=" inside a parameter

### PR DESCRIPTION
**Pull request by sashas-lb**
inaugurator changes
1) bugfix with paramter split 
2) kernel version bump for build :) 
3) adopt pika api changes
4) change label exchange to direct mode

### CODE REVIEW

File: inaugurator/ceremony.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def _getSerialDevices():
         with open("/proc/cmdline", "r") as cmdLineFile:
             cmdLine = cmdLineFile.read()
         args = cmdLine.split(" ")
-        keyValuePairs = [arg.split("=") for arg in args if "=" in arg]
+        keyValuePairs = [arg.split("=",1) for arg in args if "=" in arg]
```
Comment (maor-lb) [inline inaugurator/ceremony.py]: +1

File: Makefile.build [hunk 2]
```diff
@@ -1,4 +1,4 @@
-KERNEL_VERSION=4.11.12-200.fc25.x86_64
+KERNEL_VERSION=4.12.14-200.fc25.x86_64
```
Comment (maor-lb) [inline Makefile.build]: +1

File: inaugurator/tests/common.py [hunk 3]
```diff
@@ -61,7 +61,7 @@ def __eq__(self, other):
                     return True
             return Any()
 
-        self.exchange_declare.assert_called_with(Any(int), exchange=expectedStatusExchange, type='fanout')
+        self.exchange_declare.assert_called_with(Any(int), exchange=expectedStatusExchange, exchange_type='fanout')
```
Comment (maor-lb) [inline inaugurator/tests/common.py]: +1

File: inaugurator/talktoserver.py [hunk 4]
```diff
@@ -58,12 +58,12 @@ def _connect(self, amqpURL):
...
         self._channel.exchange_declare(exchange=self._statusExchange, exchange_type='fanout', auto_delete=True)
         logging.info("Declaring a RabbitMQ exchange %(exchange)s...", dict(exchange=self._labelExchange))
-        self._channel.exchange_declare(exchange=self._labelExchange, exchange_type='fanout', auto_delete=True)
+        self._channel.exchange_declare(exchange="inaugurator_labels", exchange_type='direct')
         logging.info("Declaring an exclusive RabbitMQ label queue...")
         frame = self._channel.queue_declare(exclusive=True)
         self._labelQueue = frame.method.queue
         logging.info("Binding label queue %(queue)s with labels exchange...", dict(queue=self._labelQueue))
-        self._channel.queue_bind(queue=self._labelQueue, exchange=self._labelExchange)
+        self._channel.queue_bind(queue=self._labelQueue, exchange="inaugurator_labels", routing_key=self._labelExchange)
```
Comment (maor-lb) [inline inaugurator/talktoserver.py]: +1M, great :)

File: inaugurator/network.py [hunk 5]
```diff
@@ -9,7 +9,7 @@ class Network:
     def __init__(self, macAddress, ipAddress, netmask, gateway):
         self._gateway = gateway
         interfacesTable = self._interfacesTable()
-        assert macAddress.lower() in interfacesTable
+        assert macAddress.lower() in interfacesTable, "macAddress %s interfacesTable %s" % (macAddress, interfacesTable)
```
Comment (sashas-lb) [inline inaugurator/network.py]: +1

**Comment by sashas-lb** [comment]
@maor-lb  please review 2 last commits in inaugurator i had tested it manually on mylaptop and osmosis worked

File: inaugurator/osmose.py [hunk 6]
```diff
@@ -35,13 +35,18 @@ def __init__(self, destination, objectStores, withLocalObjectStore, ignoreDirs,
...
     def wait(self):
+        osmosis_output = []
+        for line in iter(self._popen.stdout.readline, b''):
+            print "Osmosis:>>> %s" % line.rstrip()
+            osmosis_output.append(line)
+        self._popen.stdout.close()
         result = self._popen.wait()
         if result != 0:
-            raise Exception("Osmosis failed")
+            raise Exception("Osmosis failed: return code %d output %s" % (result, '\n'.join(osmosis_output)))
```
Comment (maor-lb) [inline inaugurator/osmose.py]: look great

File: inaugurator/osmose.py [hunk 7]
```diff
@@ -23,7 +23,8 @@ def __init__(self, destination, objectStores, withLocalObjectStore, ignoreDirs,
         elif localObjectStore is not None:
             localObjectStore = os.path.join(localObjectStore, "objectstore")
 
-        objectStores = localObjectStore + ("+" + objectStores if objectStores else "")
+        if localObjectStore is not None:
```
Comment (maor-lb) [inline inaugurator/osmose.py]: +1

**Pull request merged**